Docs-10168 Add new ALS actions to "Helpful Features in the Text Editor" (#164)
mulesoft/docs-design-center: modules/ROOT/pages/design-helpful-features-text-editor.adoc

          application/json:
            type: person  <3>
----

== Find Locations Where a Definition is Referenced

> 

Reviewer: GLOBALLY replace "click on" with just "click."

Author: Fixed

You can navigate to locations where a given definition is referenced by right-clicking on the definition and selecting *Go to References*.

> For example, the type `person` is declared on line 5 at the first callout in the following example API specification. That type is subsequently used on line 14 at the second callout, and on line 22 at the third callout. If you right-click `person` on line 5 and select *Go to References*, a list of the `person` locations (callouts two and three) appears. Double clicking on a listed location will navigate you to the relevant line.

Reviewer: Consider breaking after the second sentence, inserting the syntax (will the audience recognize these as "callouts"?), and then following the syntax with the third and fourth sentences:

"For example,...at the third callout:"
<syntax>
"If you right-click...the relevant line."

Reviewer: will navigate you > navigates you

Author: Fixed

----
#%RAML 1.0
title: Test References

uses:
  lib: library.raml 

types:
  person:  <1>
    description: this is person's description
    properties:
      firstname: string
  manager:
    type: lib.manager 
  Alertable: manager | person  <2>

/person:
  get:
    responses:
      200:
        body:
          application/json:
            type: person[]  <3>
----

> The action also supports the following:

Reviewer: What action?

Author: Fixed

* RAML resource types and traits
* RAML type expressions
* Library aliases
* References linked in JSON path navigation

> * References to nodes that apply “macros”, such as annotationTypes, securitySchemes, resourceTypes and traits

Reviewer: ```suggestion
* References to nodes that apply “macros,” such as annotationTypes, securitySchemes, resourceTypes, and traits
```
Should the macro names (?) be formatted in UI mono?

Author: Fixed

== Select Code Ranges

> “Selection Ranges” allow you to select and to expand the selection of code blocks. 

Reviewer: GLOBALLY replace "allow" with "enable," per the style ref.

Author: Fixed

> To select a range, click on an element and press Control + Shift + [Left Arrow | Right Arrow]. The Right Arrow key increases the range. The Left Arrow key decreases the range.

Reviewer: click an element

Author: Fixed

> For example, clicking on `body` in the following code, and pressing Control + Shift + Right Arrow, creates a selection range that only includes `body:`:

Reviewer: ```suggestion
For example, clicking `body` in the following code and pressing Control + Shift + Right Arrow creates a selection range that includes only `body`:
```

Author: Fixed

image:apid-helpful-text-editor-1.png[30%,30%]

Pressing Control + Shift + Right Arrow again results in the following selection range:

image:apid-helpful-text-editor-2.png[30%,30%]

> Pressing Control + Shift + Left Arrow collapses the selection range, leaving only `body:` selected again.

Reviewer: Is it `body` or `body:`? GLOBALLY search for an standardize to accurate UI presentation.

Author: Fixed

== Navigate to a Symbol

> You can right-click anywhere in the document and select *Go to Symbol*, which will present the list of defined symbols: 

Reviewer: GLOBALLY use present tense:
```suggestion
You can right-click anywhere in the document and select *Go to Symbol*, which presents the list of defined symbols: 
```
alternative
```suggestion
You can right-click anywhere in the document and select *Go to Symbol*, displaying the list of defined symbols: 
```

Author: Fixed

image:apid-helpful-text-editor-3.png[75%,75%]

> Selecting a symbol will navigate you to its location in the document.

Reviewer: navigates

Author: Fixed

== Extract an Element to a Declaration

You can take advantage of refactoring helpers to ease the continuous improvement of the design. One such helper is the ability to extract an inline declarable element to a top-level declaration, with the result being reusabiity.

When you select a range that matches an inline declarable element, an icon appears to the left of the element:

image:apid-helpful-text-editor-4.png[37%,37%]

> Tap the icon and select *Extract to Declaration*, which results in the following two changes to the document:

Reviewer: We typically use "click" instead of tap for icons; is there a particular reason for "tap" here?

Author: Fixed

image:apid-helpful-text-editor-5.png[37%,37%]

> The first callout shows that `type: lib.manager` has been replaced with `type: newtype`. A `newtype` declaration now exists outside of the `manager` scope (as shown at the second callout), thereby making `newtype` reusable.

Reviewer: has been replaced > is replaced

Author: Fixed

> The extracted declaration name is based on the original name. If the name is already being used by another declaration of the same kind, an auto increment number is added at the end of the name. You can rename the declaration.

Reviewer: an automatically incrementing number?

Author: Fixed

> Along with `types` declarations, you can also extract `resourceTypes` declarations and `traits` declarations.

Reviewer: Super clear writing, Chris! You are doing great!
